cloud/ovirt/ovirt_templates.py

```
@@ -0,0 +1,299 @@
+                        template=otypes.Template(
+                            name=module.params['name'],
+                        ),
+                        import_as_template=True,
+                    )
+                
+                if module.params['image_disk']:
+                    # We need to refresh storage domain to get list of images:
+                    templates_module._get_export_domain_service().images_service().list()
+
+                    glance_service = connection.system_service().openstack_image_providers_service()
+                    image_provider = search_by_name(glance_service, module.params['image_provider'])
+                    images_service = glance_service.service(image_provider.id).images_service()
+                else:
+                    images_service = templates_module._get_export_domain_service().templates_service()
+                template_name = module.params['image_disk'] or module.params['name'] 
+                entity = search_by_name(images_service, template_name)
+                if entity is None:
+                    raise Exception("Image/template '%s' was not found." % template_name)
+
+                images_service.service(entity.id).import_(
+                    storage_domain=otypes.StorageDomain(
+                        name=module.params['storage_domain']
+                    ) if module.params['storage_domain'] else None,
+                    cluster=otypes.Cluster(
+                        name=module.params['cluster']
+                    ) if module.params['cluster'] else None,
+                    **kwargs
+                )
+                template = wait_for_import(module, templates_service)
+                ret = {
+                    'changed': True,
+                    'id': template.id,
+                    'template': get_dict_of_struct(template),
+                }
+
+        module.exit_json(**ret)
+    except Exception as e:
+        module.fail_json(msg=str(e))
```

Same here, please surface the full exception.